flaky Error: Can't set headers after they are sent

I'm running the doiuse.com code locally to work on https://github.com/anandthakker/doiuse.com/issues/31.

I haven't found reliable steps to reproduce, but every so often, it exits with status 1 and this output:
```
_http_outgoing.js:335
    throw new Error('Can\'t set headers after they are sent.');
          ^
Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (_http_outgoing.js:335:11)
    at cssFeatures (/Users/lcrouch/code/doiuse.com/lib/css-features.js:18:7)
    at /Users/lcrouch/code/doiuse.com/index.js:46:9
    at ConcatStream.<anonymous> (/Users/lcrouch/code/doiuse.com/node_modules/mississippi/node_modules/concat-stream/index.js:36:43)
    at ConcatStream.emit (events.js:129:20)
    at finishMaybe (/Users/lcrouch/code/doiuse.com/node_modules/mississippi/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:511:14)
    at endWritable (/Users/lcrouch/code/doiuse.com/node_modules/mississippi/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:521:3)
    at ConcatStream.Writable.end (/Users/lcrouch/code/doiuse.com/node_modules/mississippi/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:486:5)
    at DestroyableTransform.onend (/Users/lcrouch/code/doiuse.com/node_modules/mississippi/node_modules/through2/node_modules/readable-stream/lib/_stream_readable.js:545:10)
    at DestroyableTransform.g (events.js:199:16)

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
npm ERR! node v0.12.2
npm ERR! npm  v2.7.4
npm ERR! code ELIFECYCLE
npm ERR! doiuse.com@1.0.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the doiuse.com@1.0.0 start script 'node index.js'.
npm ERR! This is most likely a problem with the doiuse.com package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node index.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls doiuse.com
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/lcrouch/code/doiuse.com/npm-debug.log
```

The corresponding `npm-debug.log` output:
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'start' ]
2 info using npm@2.7.4
3 info using node@v0.12.2
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info prestart doiuse.com@1.0.0
6 info start doiuse.com@1.0.0
7 verbose unsafe-perm in lifecycle true
8 info doiuse.com@1.0.0 Failed to exec start script
9 verbose stack Error: doiuse.com@1.0.0 start: `node index.js`
9 verbose stack Exit status 1
9 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:213:16)
9 verbose stack     at EventEmitter.emit (events.js:110:17)
9 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:14:12)
9 verbose stack     at ChildProcess.emit (events.js:110:17)
9 verbose stack     at maybeClose (child_process.js:1015:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
10 verbose pkgid doiuse.com@1.0.0
11 verbose cwd /Users/lcrouch/code/doiuse.com
12 error Darwin 14.5.0
13 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
14 error node v0.12.2
15 error npm  v2.7.4
16 error code ELIFECYCLE
17 error doiuse.com@1.0.0 start: `node index.js`
17 error Exit status 1
18 error Failed at the doiuse.com@1.0.0 start script 'node index.js'.
18 error This is most likely a problem with the doiuse.com package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     node index.js
18 error You can get their info via:
18 error     npm owner ls doiuse.com
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]
```

--- a/index.js
+++ b/index.js
@@ -35,15 +35,15 @@ var server = http.createServer(function (req, res) {
 
         try {
           args = JSON.parse(args)
+          cssFeatures(args, res)
         } catch (e) {
-          debug('Error parsing request ', args, e)
+          debug('Error parsing request ', e.toString(), args)
           res.statusCode = 400
           res.end(JSON.stringify({
             error: e.toString(),
             statusCode: 400
           }))
         }
-        cssFeatures(args, res)
       }),
       function (err) {
         if (err) { console.error(err) }

